src/main/java/net/sf/javaanpr/recognizer/PatternComparer.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2013 JavaANPR contributors
+ * Copyright 2006 Ondrej Martinsky
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.osedu.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package net.sf.javaanpr.recognizer;
+
+import java.util.Comparator;
+
+public class PatternComparer implements Comparator<Object> {
+    private int direction;
+
+    public PatternComparer(int direction) {
+        this.direction = direction;
+    }
+
+    @Override
+    public int compare(Object o1, Object o2) {
```

Comment:
Optional: Also, if you're up for it, the actual implementation of the `compare` method could be greatly simplified using a ternary operator + `Integer.compareTo`.

Too bad we're still on Java 6 and don't have the static `Integer.compare` (JDK7+). On a side note, what do you think about moving to Java 7/Java 8 with the project?